### PR TITLE
chore: deployer refactor

### PR DESCRIPTION
**Pull request by lucasvo**
I've hardened our deployer a bit:
1. I'm removing all arguments from the deploy methods and moving them into the constructor of the deployer. This will make it easier to verify the parameters that it was deployed with
2. I'm making sure each deploy method can only be called once.
3. Because of these two changes, we can get rid of the deploy user. The deployer does not need to authenticate anyone anymore for calling a deploy method. 
4. I'm renaming RootAdmin to TinlakeRoot

### CODE REVIEW

File: src/lender/deployer.sol [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+}
+
+contract SeniorTrancheFab {
+    string constant public symbol = "DROP";
+    string constant public name = "Tinlake Drop Token";
+    uint public ratePerSecond;
+
+    constructor (uint rate_) public {
+        ratePerSecond = rate_;
+    }
```
Comment (manuelpolzhofer) [inline src/lender/deployer.sol]: We want to re-use the Fabs on mainnet for different deployments. (with different rates)

I think each deployment will have a different token name

It should not be required to touch a solidity file if we want to deploy for another asset group.

File: src/root.sol [hunk 2]
```diff
@@ -12,53 +12,54 @@
...
-        lenderDeployer.assessor().depend("pool", borrowerPricePool);
-    }
-
-    function relyModules() internal {
-        // distributor allowed to call
-        borrowerDeployer.shelf().rely(address(lenderDeployer.distributor()));
-    }
-
-    function completeDeployment() public auth {
+    function deploy() public auth {
```
Comment (manuelpolzhofer) [inline src/root.sol]: I think we can remove the auth here.

File: src/borrower/test/mock/shelf.sol [hunk 3]
```diff
@@ -4,7 +4,7 @@ import "../../../test/mock/mock.sol";
 
 contract ShelfMock is Mock {
 
-    // maybe not needed
+    // TODO: maybe not needed
```
Comment (manuelpolzhofer) [inline src/borrower/test/mock/shelf.sol]: can be removed. 
```suggestion

```

File: src/lender/deployer.sol [hunk 4]
```diff
@@ -191,10 +191,10 @@ contract LenderDeployer is Auth {
         AuthLike(distributor).rely(root);
     }
 
-    function deploySeniorTranche() public  {
-        require(assessor != ZERO && senior == ZERO);
-        senior = seniorTrancheFab.newTranche(currency, assessor);
-        AuthLike(senior).rely(root);
+    function deployAssessor() public {
+        require(assessor == ZERO);
```
Comment (manuelpolzhofer) [inline src/lender/deployer.sol]: This should change with latest develop

File: src/borrower/deployer.sol [hunk 5]
```diff
@@ -0,0 +1,215 @@
...
+
+    function deployCeiling() public {
+        require(ceiling == ZERO);
+        ceiling = ceilingFab.newCeiling();
+        AuthLike(ceiling).rely(root);
+    }
+
+    function deploy() public {
+        // ensures all required deploy methods were called
+        require(shelf != ZERO && collector != ZERO && pricePool != ZERO);
```
[the remark this answers is not in the document]
Comment (manuelpolzhofer) [reply src/borrower/deployer.sol]: ahh, shelf transitive ensures it.

[the remark this answers is not in the document]
Comment (lucasvo) [reply src/borrower/deployer.sol]: yes I removed the transitive checks.

[the remark this answers is not in the document]
Comment (lucasvo) [reply src/borrower/deployer.sol]: (duplicate) checks

File: src/lender/deployer.sol (hunk 1 above)
Comment (lucasvo) [reply src/lender/deployer.sol]: I agree, but I didn't want to blow up this even more. I think the added security for now is better than the flexibility we had. I will try to add it in a separate PR.

**Pull request merged**